test/unittests/evmmax_test.cpp

```diff
@@ -0,0 +1,142 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2023 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <evmmax/evmmax.hpp>
+#include <gtest/gtest.h>
+#include <array>
+
+using namespace intx;
+using namespace evmmax;
+
+// TODO(intx): Add ""_u384.
+inline constexpr auto operator""_u384(const char* s)
+{
+    return intx::from_string<intx::uint384>(s);
+}
+
+constexpr auto P23 = 23_u256;
+constexpr auto BN254Mod = 0x30644e72e131a029b85045b68181585d97816a916871ca8d3c208c16d87cfd47_u256;
+constexpr auto Secp256k1Mod =
+    0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffefffffc2f_u256;
+constexpr auto M256 = 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff_u256;
+constexpr auto BLS12384Mod =
+    0x1a0111ea397fe69a4b1ba7b6434bacd764774b84f38512bf6730d2a0f6b0f6241eabfffeb153ffffb9feffffffffaaab_u384;
+
+
+template <typename UintT, const UintT& Mod>
+struct ModA : ModArith<UintT>
+{
+    using uint = UintT;
+    ModA() : ModArith<UintT>{Mod} {}
+};
+
+template <typename>
+class evmmax_test : public testing::Test
+{};
```

Comment:
Suggestion if possible: alias inside fixture class `TypeParam` as `ModA` or something close to `ModArith`.

Reply:
Maybe also could have an instance of it as class member.

Reply:
There are some limitations to this. Either you need to use `this->` or `TestFixture::`. http://google.github.io/googletest/reference/testing.html#TYPED_TEST